src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -532,23 +600,30 @@ public Solution WithProjectDocumentsOrder(ProjectId projectId, ImmutableList<Doc
         /// specified metadata reference.
         /// </summary>
         public Solution AddMetadataReference(ProjectId projectId, MetadataReference metadataReference)
-        {
-            var newState = _state.AddMetadataReference(projectId, metadataReference);
-            if (newState == _state)
-            {
-                return this;
-            }
-
-            return new Solution(newState);
-        }
+            => AddMetadataReferences(projectId, SpecializedCollections.SingletonEnumerable(
+                metadataReference ?? throw new ArgumentNullException(nameof(metadataReference))));
```

Comment:
Ditto as earlier: strange indenting.

Reply:
Checked to multi-line. BTW, this indenting is standard for multi-line expression bodies. The unfortunate consequence of `=>` not having 3 characters.